LandSandBoat/server
# Convert zone types field to a bitmask

**_I affirm:_**

## What does this pull request do?
The intention here is to pave the way towards asking the zone itself if it is a "signet zone" and the like. I thought this made more sense here than in the zone misc flags, which are more about what you can do while in a zone. here we want to know if the zone "counts" as these types, so a signet type made sense to me. Note that this is not the same as being a conquest zone. Several things related to signet will function in zones where conquest doesn't happen.
- Changed from enum class to just enum to ease bitwise maths.
- Added zonetypes for signet/sanction/sigil/ionis in the mask.
- Updated all core references to use bitwise checks. values can now be combined.
- Did update existing enum values in zone settings, but did not add new enums to the values in zone_settings yet

## Steps to test these changes
Log in and verify using prints that dungeons still count as dungeons, dynamis still counts as dynamis. There should be no observable changes before and after merging this PR.
